lib/schema.js

```diff
@@ -579,6 +580,23 @@ Schema.prototype.add = function add(obj, prefix) {
   return this;
 };
 
+Schema.prototype.removeIndex = function removeIndex(index) {
+  if (arguments.length > 1 || Object.keys(arguments[0]).length > 1 && typeof arguments[0] !== 'string') {
+    throw new Error('removeIndex() takes only 1 parameter');
+  }
+  for (let i = 0; i < this._indexes.length; i++) {
+    for (let j = 0; j < this._indexes[i].length; j++) {
```

Comment:
You don't need to loop through the index itself. You just need to check if `this._indexes[i][0]` is deep equal to the `index` argument, if the `index` argument is an object